libraries/AP_Compass/examples/AP_Compass_test/AP_Compass_test.cpp

```diff
@@ -78,7 +83,15 @@ static void loop()
 
             if (!compass.healthy()) {
                 hal.console->printf("not healthy\n");
-                continue;
+
+                max_attempts--;                 
+
+                // dont attempt at all for the first 100 iterations if not reporting healthy 
```

Comment:
This looks like it is tight-looping?  Perhaps a 100ms delay?

Reply:
... it's already in a 10Hz loop, so no it's not a 'tight loop'